docs/app/Components/ComponentDoc/ComponentDoc.js

```diff
@@ -1,31 +1,31 @@
-import React, { Component, PropTypes } from 'react'
-import { Segment } from 'stardust'
+import React, { PropTypes } from 'react'
+// import { Segment } from 'stardust'
 
 import ComponentDescription from './ComponentDescription'
 import ComponentExamples from './ComponentExamples'
 import ComponentProps from './ComponentProps'
 import getComponentDocInfo from 'docs/app/utils/getComponentDocInfo'
 
-export default class ComponentDoc extends Component {
-  static propTypes = {
-    meta: PropTypes.object,
-  };
+const ComponentDoc = ({ _meta }) => {
+  // TODO remove util in favor of separate docgen.json in each component directory
+  // this util just parses out a single docgen file based on component path name
+  const doc = getComponentDocInfo(_meta)
 
-  render() {
-    const doc = getComponentDocInfo(this.props.meta)
+  return (
+    <div>
+      <ComponentDescription
+        _meta={_meta}
+        docgen={doc.docgen}
+        docPath={doc.docPath}
+      />
+      <ComponentProps props={doc.docgen.props} />
+      <ComponentExamples name={_meta.name} />
+    </div>
+  )
```

Comment:
Updates here essentially just pass down the entire _meta object (which describes a component), whereas before it was passing bits and pieces and computing others that it shouldn't have been.